### ISSUE TITLE
Bulk Insert using repository

### ISSUE THREAD

**Issue by nicolastakashi** [rueian/rueidis#115]
Is there any way to implement Bulk Insert (Batch Insert) using a Repository?

**Comment by rueian** [comment]
Hi @nicolastakashi,

Currently there is no such API on OM Repository. However, I think it won't be too hard to implement one. Maybe I could implement next week.

For now, you can just using many goroutine to insert records concurrently. The auto-pipelining feature will batch them for you.

**Comment by rueian** [comment]
Hi @nicolastakashi,

I plan to add `SaveMulti` to `om.Repository`. Usage example:
```go
package main

import (
	"context"
	"fmt"

	"github.com/rueian/rueidis"
	"github.com/rueian/rueidis/om"
)

type obj struct {
	Key string `json:"key" redis:",key"`
	Ver int64  `json:"ver" redis:",ver"`
	Val int64  `json:"val"`
}

func main() {
	c, err := rueidis.NewClient(rueidis.ClientOption{InitAddress: []string{"127.0.0.1:6377"}})
	if err != nil {
		panic(err)
	}
	defer c.Close()

	repo := om.NewJSONRepository("obj", obj{}, c)
	fmt.Println(repo.CreateIndex(context.Background(), func(schema om.FtCreateSchema) om.Completed {
		return schema.FieldName("$.val").As("val").Numeric().Build()
	}))

	records := make([]*obj, 100000)
	for i := range records {
		records[i] = repo.NewEntity()
		records[i].Val = int64(i)
	}
	for _, err = range repo.SaveMulti(context.Background(), records...) {
		if err != nil {
			panic(err)
		}
	}

	total, records, err := repo.Search(context.Background(), func(search om.FtSearchIndex) om.Completed {
		return search.Query("*").Sortby("val").Limit().OffsetNum(0, int64(len(records))).Build()
	})
	if err != nil {
		panic(err)
	}
	if total != int64(len(records)) {
		panic("total mismatch")
	}
	for i, r := range records {
		if r.Ver != 1 {
			panic("ver mismatch")
		}
		if r.Val != int64(i) {
			panic("val mismatch")
		}
	}
}

```
What do you think?

**Comment by nicolastakashi** [comment]
Not bad at all!
This gives all control to the user.
Should we provide any perf comparison in the and?

**Comment by rueian** [comment]
Hi @nicolastakashi,

Here is an example benchmark on my intel macbook:
```go
package main

import (
	"context"
	"sync"
	"testing"

	"github.com/rueian/rueidis"
	"github.com/rueian/rueidis/om"
)

type obj struct {
	Key string `json:"key" redis:",key"`
	Ver int64  `json:"ver" redis:",ver"`
	Val int64  `json:"val"`
}

func BenchmarkName(b *testing.B) {
	c, err := rueidis.NewClient(rueidis.ClientOption{InitAddress: []string{"127.0.0.1:6377"}})
	if err != nil {
		panic(err)
	}
	defer c.Close()

	setup := func() (om.Repository[obj], []*obj) {
		repo := om.NewJSONRepository("obj", obj{}, c)
		records := make([]*obj, 10000)
		for i := range records {
			records[i] = repo.NewEntity()
			records[i].Val = int64(i)
		}
		return repo, records
	}

	b.Run("SaveMulti", func(b *testing.B) {
		repo, records := setup()
		for i := 0; i < b.N; i++ {
			for _, err = range repo.SaveMulti(context.Background(), records...) {
				if err != nil {
					panic(err)
				}
			}
		}
	})
	b.Run("Sequential Save", func(b *testing.B) {
		repo, records := setup()
		for i := 0; i < b.N; i++ {
			for _, record := range records {
				if err = repo.Save(context.Background(), record); err != nil {
					panic(err)
				}
			}
		}
	})
	b.Run("Concurrent Save", func(b *testing.B) {
		repo, records := setup()
		for i := 0; i < b.N; i++ {
			var wg sync.WaitGroup
			wg.Add(len(records))
			for _, record := range records {
				go func(record *obj) {
					if err = repo.Save(context.Background(), record); err != nil {
						panic(err)
					}
					wg.Done()
				}(record)
			}
			wg.Wait()
		}
	})
}
```

```shell
▶ go test -bench=. ./cmd/json/...
goos: darwin
goarch: amd64
pkg: github.com/rueian/rueidis/cmd/json
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkName/SaveMulti-12         	       4	 262156222 ns/op
BenchmarkName/Sequential_Save-12   	       1	14287701904 ns/op
BenchmarkName/Concurrent_Save-12   	       4	 274334126 ns/op
PASS
ok  	github.com/rueian/rueidis/cmd/json	19.686s
```

**Comment by nicolastakashi** [comment]
Amazing improvement Dude!
You rock 💪🏼

**Comment by rueian** [comment]
Hi @nicolastakashi,
`SaveMulti` has already been released in v0.0.83.
